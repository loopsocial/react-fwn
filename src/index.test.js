import React from 'react'
import { render } from '@testing-library/react'
import ReactFWN from './index'

const headAppendChildMock = jest.spyOn(document.head, 'appendChild')

describe('ReactFWN component', () => {
  it("should inject component's script", () => {
    const { rerender, unmount } = render(<ReactFWN />)

    unmount()
    rerender(<ReactFWN />)

    unmount()
    rerender(<ReactFWN />)

    expect(headAppendChildMock.mock.calls).toMatchSnapshot()
  })

  it('should render default <fw-embed-feed /> custom component', () => {
    const { container, getByText } = render(
      <ReactFWN arg1="value1" arg2="value2" />
    )

    getByText((_, element) => element.tagName.toLowerCase() === 'fw-embed-feed')
    expect(container).toMatchSnapshot()
  })

  test('should render <fw-storyblock /> custom component for mode=story', () => {
    const { container, getByText } = render(
      <ReactFWN mode="story" arg1="value1" arg2="value2" />
    )

    getByText((_, element) => element.tagName.toLowerCase() === 'fw-storyblock')
    expect(container).toMatchSnapshot()
  })
})
