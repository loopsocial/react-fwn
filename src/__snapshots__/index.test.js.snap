// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactFWN component should inject component's script 1`] = `
Array [
  Array [
    <script
      src="https://asset.fwcdn2.com/js/fwn.js"
      type="text/javascript"
    />,
  ],
]
`;

exports[`ReactFWN component should render <fw-storyblock /> custom component for mode=story 1`] = `
<div>
  <fw-storyblock
    arg1="value1"
    arg2="value2"
    mode="story"
  />
</div>
`;

exports[`ReactFWN component should render default <fw-embed-feed /> custom component 1`] = `
<div>
  <fw-embed-feed
    arg1="value1"
    arg2="value2"
  />
</div>
`;
